topn-limit-push-down.md

```diff
@@ -0,0 +1,124 @@
+---
+title: TopN 和 Limit 下推
+category: performance
+---
+
+# TopN 和 Limit 下推
+
+SQL 中的 LIMIT 子句在 TiDB 查询计划树中对应 Limit 算子节点，ORDER BY 子句在查询计划树中对应 Sort 算子节点，此外，我们会将相邻的 Limit 和 Sort 算子组合成 TopN 算子节点，表示按某个排序规则提取记录的前 N 项。从另一方面来说，Limit 节点等价于一个排序规则为空的 TopN 节点。
+
+和谓词下推类似，TopN（及 Limit，下同）下推将查询计划树中的 TopN 计算尽可能下推到距离数据源最近的地方，以尽早完成数据的过滤，进而显著地减少数据传输或计算的开销。
+
+如果要关闭这个规则，可参照[优化规则及表达式下推的黑名单](/blacklist-control-plan.md)中的关闭方法。
```

Comment:
/blacklist-control-plan.md is not added to release-2.1 branch, so please decide whether you need to keep this link or not. @LittleFall